packages/realm/src/app-services/Sync.ts

```diff
@@ -60,7 +60,7 @@ export class Sync {
 
   /** @deprecated Will be removed in v13.0.0. Please use {@link Realm.setLogger}. */
   static setLogger(app: App, logger: Logger) {
-    const factory = binding.Helpers.makeLoggerFactory((level, message) => {
+    const factory = binding.Helpers.makeLoggerFactory((category, level, message) => {
```

Comment:
The logger received to `Sync.setLogger()` is currently (in our TS types) the one taking the numeric log level (thus, same as before). Core hasn't changed that API right?
```suggestion
    const factory = binding.Helpers.makeLoggerFactory((_, level, message) => {
```